pkgs/build-support/coq/default.nix

```
@@ -1,10 +1,33 @@
 { lib, stdenv, coqPackages, coq, which, fetchzip }@args:
-let lib = import ./extra-lib.nix {inherit (args) lib;}; in
-with builtins; with lib;
+
 let
+  lib = import ./extra-lib.nix {
+    inherit (args) lib;
```

Ew, that's bound to confuse everyone but Coq'ers. Why not:

```suggestion
  extra-lib = import ./extra-lib.nix {
    inherit (args) lib;
```

And then explicitly inherit from `extra-lib` below? That's exactly what this general sort of change allows now.

---

Funny -- that's exactly what I started with, then moved to this to imitate the previous naming.

In a very real sense, since `extra-lib` uses `recursiveUpdate lib`, this _is_ the nixpkgs lib, just with some updates. I'm not a fan of the pattern, since it means that there's two things going under the same name, but it appears to be the way the Coq section of nixpkgs wants to do things.

If I do the transformation from `lib` to `extra-lib`, every use of `lib` below here needs to change to `extra-lib`. In my eyes, that motivates keeping the overlapping name.